File: apps/python/blueprints/requests.py
import azure.functions as func
import json
import logging

bp = func.Blueprint()
logger = logging.getLogger(__name__)


@bp.route(route="requests/log-levels", methods=["GET"])
def log_levels(req: func.HttpRequest) -> func.HttpResponse:
    """Demonstrate logging at all severity levels."""
    logger.info("Log levels endpoint requested")

    levels = ["DEBUG", "INFO", "WARNING", "ERROR", "CRITICAL"]

    logger.debug("Debug level log message")
    logger.info("Info level log message")
    logger.warning("Warning level log message")
    logger.error("Error level log message")
    logger.critical("Critical level log message")

    body = json.dumps({
        "message": "Logged at all levels",
        "levels": levels,
    })
    return func.HttpResponse(body, mimetype="application/json", status_code=200)

File: apps/nodejs/src/functions/requests.js
const { app } = require('@azure/functions');

/**
 * Demonstrate logging at all severity levels.
 */
app.http('logLevels', {
    methods: ['GET'],
    route: 'requests/log-levels',
    handler: async (request, context) => {
        context.log('Log levels endpoint requested');

        const levels = ['DEBUG', 'INFO', 'WARNING', 'ERROR', 'CRITICAL'];

        context.trace('Debug level log message');
        context.log('Info level log message');
        context.warn('Warning level log message');
        context.error('Error level log message');

        return {
            status: 200,
            jsonBody: {
                message: 'Logged at all levels',
                levels,
            },
        };
    },
});
